syntax = "proto2";
message user_registered {
    required uint64 user_id = 1;
    required string name = 2;
}

message user_renamed {
    required uint64 user_id = 1;
    required string name = 2;
}

message user_deal_won {
    required uint64 user_id = 1;
    required string time = 2;
    required int64 amount = 3;
}

message user_connected {
    required uint64 user_id = 1;
}

message user_disconnected {
    required uint64 user_id = 2;
}

message wrapper_message {
    oneof msg {
        user_registered user_registered_msg = 1;
        user_renamed user_renamed_msg = 2;       
	user_deal_won user_deal_won_msg = 3;
	user_connected user_connected_msg = 4;
	user_disconnected user_disconnected_msg = 51;
    }
}